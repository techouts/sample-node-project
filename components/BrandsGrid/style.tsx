import Grid from "@mui/material/Grid";
import Button from "@mui/material/Button";
import Typography from "@mui/material/Typography";
import Box from "@mui/material/Box";
import styledComponent from "@emotion/styled";
import { styled } from "@mui/material/styles";

interface imageRespose {
  imageSource: any;
}

interface backgroundimageInterface {
  imageSource: any;
  isBackgroundExist: boolean;
}
export const MainBox = styled(Box)(({ imageSource }: imageRespose) => ({
  display: "flex",
  flexDirection: imageSource ? "row" : "column",
  "@media(max-width:600px)": {
    flexDirection: imageSource ? "column" : "",
  },
}));
export const Title = styled(Typography)({
  fontSize: "32px",
  fontWeight: 700,
  lineHeight: "140%",
  textAlign: "center",
  padding: "2.8% 0 3.2% 0",
  "@media(max-width:600px)": {
    fontSize: "16px",
    padding: "14% 0 7% 0",
  },
});
export const ContentGrid = styled(Grid)(
  ({ imageSource, isBackgroundExist }: backgroundimageInterface) => ({
    position: isBackgroundExist ? "absolute" : "relative",
    gridTemplateColumns: imageSource ? "repeat(3, 1fr)" : "repeat(6, 1fr)",
    display: "grid",
    width: "100%",
    rowGap: imageSource ? "8%" : "9%",
    columnGap: imageSource ? "4.5%" : "1.4%",
    justifyItems: "center",
    alignItems: "center",
    padding: imageSource ? "3.5% 6.5% 3.5% 5.5%" : "4.4% 2.7%",
    "@media(max-width:600px)": {
      gridTemplateColumns: "repeat(3, 1fr)",
      rowGap: imageSource ? "15%" : "7px",
      columnGap: imageSource ? "4.5%" : "7px",
      padding: imageSource ? "7.5% 4.5%" : "17px",
    },
  })
);
export const GridItemsImage = styled(Box)({
  display: "grid",
  width: "100%",
});
export const IconImages = styledComponent.img`
width:100%;
cursor:pointer;
`;
export const StyledButton = styled(Button)(({ imageSource }: imageRespose) => ({
  backgroundColor: imageSource ? "#DEA3B7 !important" : "black !important",
  color: imageSource ? "#000" : "white",
  boxShadow: "none",
  borderRadius: "0",
  cursor: "pointer",
  marginTop: "2.8%",
  fontSize: "11px",
  fontWeight: "500",
  "&.MuiButtonBase-root.MuiButton-root": {
    padding: imageSource ? "6px 18px" : "20px 50px",
  },
  "@media(max-width:600px)": {
    marginTop: imageSource ? "3.5%" : "11%",
    "&.MuiButtonBase-root.MuiButton-root": {
      padding: imageSource ? "6px 18px" : "15px 56px",
    },
  },
}));
export const Backgroundimage = styledComponent.img`
 width: 100%;
 height: 100%;
`;
export const Firstimage = styledComponent.img`
 width: 100%;
 height: 100%;
`;
export const ImageBox = styled(Box)({
  position: "relative",
  display: "flex",
  justifyContent: "center",
  alignContent: "center",
});
