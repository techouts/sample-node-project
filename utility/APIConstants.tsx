const NEXT_PUBLIC_JUSPAY_APIURL = process.env.NEXT_PUBLIC_JUSPAY_APIURL;
export const merchant_id = "ssbeauty";
export const oauth_token = "authorizationserver/oauth/token";
export const otp_services = "services/v2_1/ssl/sendOTP/OB";
export const otp_service_verifier =
  "services/v2_1/ssl/users/loginOrSignup?mobileNumber=9640558040&otp=782253&cliend_id=mobile_android";
export const SignUp_client_id = "mobile_android";
export const getPaymentMethods = `${NEXT_PUBLIC_JUSPAY_APIURL}/merchants/${merchant_id}/paymentmethods?options.add_emandate_payment_methods=false&options.check_wallet_direct_debit_support=false&options.add_supported_reference_ids=false`;
export const CardBin = `${NEXT_PUBLIC_JUSPAY_APIURL}/cardbins/`;
export const CardTokenization = `${NEXT_PUBLIC_JUSPAY_APIURL}/card/tokenize`;
export const CardTransaction = `${NEXT_PUBLIC_JUSPAY_APIURL}/txns`;
export const QyPaymentStatus = `${NEXT_PUBLIC_JUSPAY_APIURL}/order/payment-status?merchant_id=ssbeauty&order_id=`;
export const paymentStatus =
  "/order/payment-status?merchant_id=ssbeauty&order_id=";
export const GOOGLE_MAP_DIRECTIONS_API = (
  origin: string,
  destination: string
) =>
  `${process.env.NEXT_PUBLIC_GOOGLE_MAP_URL}?api=1&origin=${origin}&destination=${destination}&travelmode=car`;
  export const UNBXD_HOST = process.env.NEXT_UNBXD_HOST;

  export const IS_UNBXD_ENABLED = process.env.NEXT_PUBLIC_IS_UNBXD_ENABLED === "true";

  export const MIDDLEWARE_UNBXD = process.env.NEXT_PUBLIC_MIDDLEWARE_UNBXD_SEARCH;

  export const IS_UNBXD_MIDDLEWARE_ENABLED = process.env.NEXT_PUBLIC_IS_UNBXD_MIDDLEWARE_ENABLED === "true"

  const NEXT_PUBLIC_D365_APP_ID_GUEST = process.env.NEXT_PUBLIC_D365_APP_ID_GUEST;

  const NEXT_PUBLIC_D365_APP_ID_LOGGED_IN = process.env.NEXT_PUBLIC_D365_APP_ID_LOGGED_IN;

  const NEXT_PUBLIC_D365_ORG_ID = process.env.NEXT_PUBLIC_D365_ORG_ID;
  
  const NEXT_PUBLIC_D365_ORG_URL = process.env.NEXT_PUBLIC_D365_ORG_URL;
  export const Google_API_KEY = "AIzaSyCBo5PdGTw5Z0BS6hTboacbxkMC7vIZ6FY";

  export {NEXT_PUBLIC_D365_APP_ID_LOGGED_IN, NEXT_PUBLIC_D365_APP_ID_GUEST, NEXT_PUBLIC_D365_ORG_ID, NEXT_PUBLIC_D365_ORG_URL}
  export const IS_RECAPTCHA_ENABLED = process.env.NEXT_PUBLIC_IS_RECAPTCHA_ENABLED === "true";