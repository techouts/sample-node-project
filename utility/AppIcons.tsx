export const Camera_ICON = "Camera_icon";
export const EDIT_IMAGE_ICON = "EDIT_IMAGE";
export const InfoCircle_Grey_ICON = "InfoCircle_Grey";
export const Profile_Add_ICON = "Profile_Add";

export const TIER0_ICON = "TIER0_LOGO";
export const TIER1_ICON = "TIER1_LOGO";
export const TIER2_ICON = "TIER2_LOGO";
export const TIER3_ICON = "TIER3_LOGO";
export const TIER4_ICON = "TIER4_LOGO";

export const GET_APP_LOGO_HEADER = "Get_App";
export const GIFT_CARD_ICON = "Gift_Card";
export const STORES_AND_EVENTS_ICON = "Stores_And_Events";
export const HELP_ICON = "Help";
export const USER_SIGNIN_ICON = "user_signin";
export const LOCATION_ICON = "Location";
export const DOWN_ARROW_ICON = "ARROWDOWN";
export const DELETE_ICON = "Delete_icon";
export const ADD_ADDRESS_ICON = "Add_address";
export const CHECKED_ICONS = "CHECKED_ICON";
export const UNCHECKED_ICONS = "UNCHECKED_ICON";
export const BeautyProfilecheck_ICON = "BeautyProfilecheck";
export const EDIT_PINK_ICON = "EditIcon_PinkColour";
export const CHECKED_RADIO_ICON = "Selected_Radio_Button";
export const UNCHECKED_RADIO_ICON = "Unselected_Radio_Button";
export const VERIFIED_IMAGE_ICON = "VERIFIED_IMAGE";
export const TICK_CIRCLE_WARNING = "TickCircleWarning";
export const COD_ICON = "CODImage";
export const CART_TRUCK_ICON = "CartTruckImage";
export const EMPTY_WISHLIST_COLOUR = "Wishlist_Empty_Colour";
export const NORMAL_SEARCH_ICON = "SEARCH_NORMAL";
export const FILTER_SETTINGS_ICON = "Filter_settings";
export const REFRESH_ICONS = "Refresh_icon";
export const SPEAKER_ICONS = "Speaker_icon";
export const CARD_PAYMENT_IMAGE = "CardPaymentImage";
export const AIRTEL_MOBILE_IMAGE = "Airtel_image_mobile";
export const MOBIWICK_MOBILE_IMAGE = "Mobiwick_mobile_image";
export const FREECHARGE_MOBILE_IMAGE = "Freecharge_mobile_image";
export const JIO_MOBILE_IMAGE = "Jio_mobile_image";
export const ZOOM_ICON = "ZoomIconUrl";
export const SHARE_ICONS = "Share_Icon";
export const WISHLIST_ADDED_ICON = "Wishlist_filled_heart";
export const PLAY_VIDEO_ICON = "VideoPlayButton";
export const LEFT_ARROW_ICON = "Left_direction_icon";
export const USER_ACCOUNT_ICON = "account_user_icon"
export const CLOSE_CIRCLE_ICON = "close-circle";
export const LIKE_ICONS = "LIKE_ICON";
export const UNLIKE_ICONS = "UNLIKE_ICON";
export const FILLED_TICK_CIRLE_ICON = "Filled_tick_circle";
export const Closecircle_BlackColour_ICON = "Closecircle_BlackColour";
export const current_location_pointer_ICON = "current_location_pointer";
export const Empty_wishlist_webicon_icon = "Empty_wishlist_webicon";
export const Empty_wishlist_appbgicon_icon = "Empty_wishlist_appbgicon";
export const CLOSE_CIRCLE_IMAGE = "close_circle";
export const DISCOUNTS_ICON = "discount_shape";
export const SORT_ICON = "SORT_ARROW";
export const EMPTY_STAR_ICON = "EmptyStar";
export const TROPHY_ICONS = "Trophy_icon";
export const GPS_LOCATION_ICON = "Gps_Pink_Icon";
export const HEADER_SEARCH_ICON = "headerSearchIcon";
export const HEADER_WISHLIST_ICON = "headerWishlistIcon";
export const HEADER_CART_ICON = "headerCartIcon";
export const BRAND_UP_ARROW = "BRAND_UP_ARROW";
export const BRAND_DOWN_ARROW = "BRAND_DOWN_ARROW";
export const HEADER_BACK_ARROW = "headerNavigationIcon";
export const INFO_ICON = "INFO_ICON";
export const FALL_BACK_IMAGE_PRODUCT = "PRODUCT_FALLBACK_URL";
export const Error_Image = "ErrorImage";
export const No_RESULTS_SEARCH_ICON = "noResultsSearchIcon";
export const RECENT_SEARCH_ICON =
  "https://storage.googleapis.com/images_cms_sit_ssb/recent_clock_bde48d7c4e_1_986324a98d.svg";