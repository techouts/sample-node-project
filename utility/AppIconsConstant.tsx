import { useRecoilState } from "recoil";
import { SSBLogos } from "../recoilstore";

export const AppIcons = (image: string) => {
  const [SSBeautyLogos] = useRecoilState(SSBLogos);

  const imageIcon = SSBeautyLogos?.appLogos?.find(
    (item: any) => item?.name == image
  );
  return imageIcon;
};
