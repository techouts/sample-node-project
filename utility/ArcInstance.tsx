import axios from "axios";
import { Cookies } from "react-cookie";

const cookie = new Cookies();
const NEXT_PUBLIC_ORC_LAYER = process.env.NEXT_PUBLIC_ORC_LAYER;

export const arcClient = axios.create({
  baseURL: NEXT_PUBLIC_ORC_LAYER,
});

arcClient.interceptors.request.use((config) => {
  const accessToken = `${cookie.get("accessToken")}`;
  const axiosConfig = { ...config };
  axiosConfig.headers["Content-Type"] = "application/json";
  axiosConfig.headers["Access-Control-Allow-Origin"] = "*";
  axiosConfig.headers["Accept"] = "*/*";
  axiosConfig.headers["Accept-Encoding"] = "gzip, deflate, br";
  axiosConfig.headers["Connection"] = "keep-alive";
  if (accessToken && accessToken !== "undefined") {
    axiosConfig.headers.Authorization = `Bearer ${accessToken}`;
  }
  return axiosConfig;
});