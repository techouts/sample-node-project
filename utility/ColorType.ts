import { type } from "os";

type RGB = `rgb(${number}, ${number}, ${number})`;
type RGBA = `rgba(${number}, ${number}, ${number}, ${number})`;
type HEX = `#${string}`;

export type ColorType = RGB | RGBA | HEX;