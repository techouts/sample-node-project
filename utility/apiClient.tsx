import axios, { AxiosRequestConfig, AxiosResponse } from "axios";
import { callErrorEvent } from "./ErrorEventAnalytics";
import { fetchMiddleWareAccessToken } from "./MiddelwareAccessWrapper";

const GRAPHQL_URL = process.env.NEXT_PUBLIC_GRAPHQL_URL!;
const MIDDLEWARE_URL = process.env.NEXT_PUBLIC_SSO_API_GATEWAY_URL!;
const isMiddlewareSecurityEnabled =
  process.env.NEXT_PUBLIC_IS_MIDDLEWARE_SECURITY_ENABLED === "true";

const axiosInstance = axios.create();

async (config: AxiosRequestConfig<any>) => {
  if (!config.url?.includes("/middleware")) return config;

  const token = localStorage.getItem("accessToken");
  const updatedHeaders: AxiosRequestConfig["headers"] = {
    ...config.headers,
    authorization: token ? `Bearer ${token}` : "",
  };
  if (isMiddlewareSecurityEnabled) {
    let mToken = localStorage.getItem("m-token");
    if (mToken && mToken !== "") {
      updatedHeaders["x-auth-token"] = `Bearer ${mToken}`;
    } else {
      const mTokenRes = await fetchMiddleWareAccessToken();
      if (mTokenRes?.data?.access_token) {
        updatedHeaders[
          "x-auth-token"
        ] = `Bearer ${mTokenRes?.data?.access_token}`;
      }
    }
  }

  return { ...config, headers: updatedHeaders };
};

axiosInstance.interceptors.response.use(
  (response: AxiosResponse) => {
    return response;
  },
  async (error: any) => {
    if (error.response) {
      const { message, status } = error.response.data || {};
      console.error(`[Error]: ${message}, Status: ${status}`);

      let params = {
        error_message: message || error.message,
        status: status || error.response?.status,
        item_name: GRAPHQL_URL,
      };

      callErrorEvent(params);
    } else if (error.request) {
      console.error("[Network error]: No response received");
    } else {
      console.error(`[Error]: ${error.message}`);
    }
    return Promise.reject(error);
  }
);

// Retry logic for requests
const retryRequest = async (
  config: AxiosRequestConfig<any>,
  retries = 3
): Promise<AxiosResponse | void> => {
  let lastError;

  for (let i = 0; i < retries; i++) {
    try {
      return await axiosInstance(config);
    } catch (error: any) {
      lastError = error;

      if (error.response?.status === 401 || error.response?.status === 403) {
        console.warn(
          `Attempt ${i + 1}: Received ${
            error.response.status
          }, trying to refresh token...`
        );
        const mTokenRes = await fetchMiddleWareAccessToken();
        const mToken = mTokenRes?.data?.access_token;

        if (!mToken) {
          console.error("Token refresh failed. Aborting retries.");
          break;
        }
        localStorage.setItem("m-token", mToken);
        config = {
          ...config,
          headers: {
            ...(config.headers || {}),
            "x-auth-token": `Bearer ${mToken}`,
          },
        };

        continue;
      }

      console.error(
        `Request failed with status ${error.response?.status}. Retrying...`
      );
    }
  }

  return Promise.reject(lastError);
};

// Function to send requests
const sendRequest = async (
  query: string,
  isCMSQuery = false,
  middleWareURL = false,
  urlKey?: string
): Promise<AxiosResponse | void> => {
  const url = urlKey
    ? `${urlKey}?query=${query}`
    : isCMSQuery
    ? query
    : middleWareURL
    ? `${MIDDLEWARE_URL}?query=${query}`
    : `${GRAPHQL_URL}?query=${query}`;

  const config: AxiosRequestConfig = {
    method: "get",
    url: url,
    headers: {},
  };

  try {
    return await retryRequest(config);
  } catch (error: any) {
    console.error(`Request failed: ${error}`);
    return Promise.reject(error);
  }
};

export default sendRequest;
