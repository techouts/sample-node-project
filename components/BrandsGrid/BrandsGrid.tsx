import { Box } from "@mui/material";
import { useEffect, useRef, useState } from "react";
import BrandsGridInterface from "../../schemas/BrandsGridSchema";
import { useMobileCheck } from "../../utility/isMobile";
import { ReplaceImage } from "../../utility/ReplaceImage";
import {
  MainBox,
  ContentGrid,
  ImageBox,
  Backgroundimage,
  Firstimage,
  GridItemsImage,
  IconImages,
  StyledButton,
  Title,
} from "./style";
import { widget_type, event_type } from "../../utility/GAConstants";
import triggerGaViewPromotion from "../../utility/GAViewPromotion";
import { Cookies } from "react-cookie";
import useStorage from "../../utility/useStoarge";

const BrandsGrid = ({
  bgColor,
  bgPadding,
  contentImageUrl,
  contentImageUrlMobile,
  contentImagePath,
  bgImageUrl,
  bgMobileImageUrl,
  items,
  button,
  position,
  title,
  id,
  __component,
  promotion_id,
  promotion_name,
}: BrandsGridInterface) => {
  const [visibleCount, setVisibleCount] = useState(0);
  const [firstNonZeroDiff, setFirstNonZeroDiff] = useState<number | null>(null);
  const isMobile = useMobileCheck();
  const gridItemRefs = useRef([]);
  const cookie = new Cookies();
  const { getItem } = useStorage();

  useEffect(() => {
    const observer = new IntersectionObserver(
      (entries) => {
        let newVisibleCount = 0;
        entries.forEach((entry) => {
          if (entry.isIntersecting) {
            const index = parseInt(entry.target.dataset.index, 10);
            newVisibleCount = Math.max(newVisibleCount, index + 1);
          }
        });
        setVisibleCount((prev) => {
          const diff = newVisibleCount - prev; 
          if (diff > 0 && firstNonZeroDiff === null) {
            setFirstNonZeroDiff(diff); 
          }
  
          return Math.max(prev, newVisibleCount);
        });
      },
      { threshold: 0.25 }
    );

    // Observe each grid item
    gridItemRefs.current.forEach((ref) => {
      if (ref) observer.observe(ref);
    });

    return () => {
      gridItemRefs.current.forEach((ref) => {
        if (ref) observer.unobserve(ref);
      });
    };
  }, [items]);

  useEffect(() => {
    if (visibleCount > 0  && items?.length > 0) {
      const startIndex = (visibleCount - firstNonZeroDiff)
      const endIndex = visibleCount;
      const datasLayer = {
        previous_screen_name: "na",
        visitor_type: cookie.get("accessToken") ? "loggedin" : "Guest",
        user_id: getItem("customer_ref", "local") || "na",
        creative_name: `${items
          ?.slice(startIndex, endIndex)
          .map((item) => `${item?.creative_name}-outer BRG`)
          .join("|")}`,
        creative_slot: `${items
          ?.slice(startIndex, endIndex)
          .map((item) => `${item?.creative_slot}-outer BRG`)
          .join("|")}`,
          ecommerce: {
            items: items
            ?.slice(startIndex, endIndex)
            .map((item, index) => ({
              promotion_id: `${promotion_id}-inner BRG`,
              promotion_name: `${promotion_id}-inner BRG`,
              creative_name: `${item?.creative_name}-inner BRG`,
              creative_slot: `${item?.creative_slot}-inner BRG`,
            })),
          }
      };
      
      triggerGaViewPromotion("view_promotion", datasLayer);
    }
  }, [ visibleCount, isMobile , firstNonZeroDiff]);

  const handleClick = (url: string,  item: any , index?: any) => {
    if (url) {
      const datasLayer = {
        previous_screen_name: "na",
        visitor_type: cookie.get("accessToken") ? "loggedin" : "Guest",
        user_id: getItem("customer_ref", "local") || "na",
        creative_name: `${items?.creative_name}-outer BRD`,
        creative_slot: `${items?.creative_slot}-outer BRD`,
        item_id: `${__component}_${position}_${index + 1}`,
        item_name: "na",
        item_type: "na",
        widget_type: widget_type,
        widget_description: "na",
        widget_position: position,
        link_url: `${global?.window?.location?.origin}${url}`,
        link_text: "na",
        no_of_items: items?.length,
        index: index + 1,
        item_brand: "na",
        item_category: "na",
        item_category2: "na",
        item_category3: "na",
        item_original_price: 0,
        item_price: 0,
        item_rating: "na",
        event_type: event_type,
        item_image_link: contentImageUrl,
        widget_title: __component,
        ecommerce: {
        items: [
          {
            promotion_id: `${promotion_id}-inner BRD`,
            promotion_name: `${promotion_id}-inner BRD`,
            creative_name: `${item?.creative_name}-inner BRD`,
            creative_slot: `${item?.creative_slot}-inner BRD`,
          }
        ]
      }
      };
      isMobile ? window.location.assign(url) : window.open(url);
      triggerGaViewPromotion("select_promotion", datasLayer);
    }
  };
  return (
    <>
      <MainBox
        imageSource={contentImageUrl && contentImageUrlMobile}
        p={isMobile ? "10px 16px" : bgPadding}
        bgcolor={bgColor}
      >
        {contentImageUrl && contentImageUrlMobile
          ? ""
          : title && <Title>{title}</Title>}
        {contentImageUrl && contentImageUrlMobile && (
          <Box display="flex" sx={{cursor: contentImagePath ? "pointer" : "default"}}>
            <Firstimage
              onClick={() => handleClick(contentImagePath)}
              src={
                isMobile
                  ? `${ReplaceImage(contentImageUrlMobile)}`
                  : `${ReplaceImage(contentImageUrl)}`
              }
            ></Firstimage>
          </Box>
        )}
        <ImageBox>
          {bgMobileImageUrl && bgImageUrl && (
            <Backgroundimage
              src={
                isMobile
                  ? `${ReplaceImage(bgMobileImageUrl)}`
                  : `${ReplaceImage(bgImageUrl)}`
              }
              alt="background image"
            ></Backgroundimage>
          )}
          <ContentGrid
            imageSource={contentImageUrl && contentImageUrlMobile}
            isBackgroundExist={bgImageUrl ? true : false}
          >
            {items?.map((item, index) => (
              <GridItemsImage
                key={index}
                data-index={index}
                ref={(el) => (gridItemRefs.current[index] = el)}
              >
                <IconImages
                  onClick={() => handleClick(item?.path, item, index)}
                  src={
                    isMobile
                      ? `${ReplaceImage(item?.imageUrlMobile)}`
                      : `${ReplaceImage(item?.imageUrl)}`
                  }
                />
              </GridItemsImage>
            ))}
          </ContentGrid>
        </ImageBox>
        <Box display="flex" justifyContent="center">
          {isMobile && button && (
            <StyledButton
              onClick={() => handleClick(contentImagePath)}
              variant="contained"
              imageSource={contentImageUrl && contentImageUrlMobile}
            >
              {button}
            </StyledButton>
          )}
          {isMobile || (contentImageUrl && contentImageUrlMobile)
            ? ""
            : button && (
                <StyledButton
                  onClick={() => handleClick(contentImagePath)}
                  variant="contained"
                  imageSource={contentImageUrl && contentImageUrlMobile}
                >
                  {button}
                </StyledButton>
              )}
        </Box>
      </MainBox>
    </>
  );
};
export default BrandsGrid;
