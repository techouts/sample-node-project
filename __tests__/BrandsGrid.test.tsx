import { render } from "@testing-library/react";
import { ColorType } from "../utility/ColorType";
import BrandsGrid from "../components/BrandsGrid/BrandsGrid";
const color: ColorType = "#0000";

const brandsGridData = {
  id: 3,
  __component: "widget.brands-grid",
  bgColor: color,
  bgPadding: "3% 5%",
  contentImageUrl:
    "https://ssbeauty-cms.s3.ap-south-1.amazonaws.com/Top_Brands_dcc7dddfe5.png",
  contentImageUrlMobile:
    "https://s3.ap-south-1.amazonaws.com/images-cms.uat.shopper-stop.in/Top_Brands_Mobile_84f472044f.png?updated_at=2022-10-03T14:23:49.866Z",
  contentImagePath: "/brand/faces-canada",
  bgImageUrl:
    "https://ssbeauty-cms.s3.ap-south-1.amazonaws.com/lipsticksgroup_12993b9cc1.png",
  bgMobileImageUrl:
    "https://ssbeauty-cms.s3.ap-south-1.amazonaws.com/Top_Banrdsbg_Image_Msite_66128a17c8.png",
  button: "EXPLORE BRANDS",
  contentImageisNewTab: false,
  display: null,
  title: "",
  buttonPath: "",
  items: [
    {
      id: 19,
      imageUrl:
        "https://ssbeauty-cms.s3.ap-south-1.amazonaws.com/Estee_Lauder_b8afab62ba.png",
      imageUrlMobile:
        "https://s3.ap-south-1.amazonaws.com/images-cms.uat.shopper-stop.in/Smashbox_bf109553e7.jpg",
      path: "/brand/faces-canada",
      isNewTab: false,
    },
  ],
  position: 6,
};

describe("Brands Grid", () => {
  it("renders a brands grid component", () => {
    const { container } = render(<BrandsGrid {...brandsGridData} />);
    expect(container).toMatchSnapshot();
  });
});
