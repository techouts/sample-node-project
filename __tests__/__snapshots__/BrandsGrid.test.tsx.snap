// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brands Grid renders a brands grid component 1`] = `
<div>
  <div
    class="MuiBox-root css-10grsak"
    imagesource="https://s3.ap-south-1.amazonaws.com/images-cms.uat.shopper-stop.in/Top_Brands_Mobile_84f472044f.png?updated_at=2022-10-03T14:23:49.866Z"
  >
    
    <div
      class="MuiBox-root css-11dqbsu"
    >
      <img
        class="css-idoo5b"
        src="https://images-cms.uat.shopper-stop.in/Top_Brands_Mobile_84f472044f.png"
      />
    </div>
    <div
      class="MuiBox-root css-xby94j"
    >
      <img
        alt="background image"
        class="css-idoo5b"
        src="https://images-cms.sit.shopper-stop.in/Top_Banrdsbg_Image_Msite_66128a17c8.png"
      />
      <div
        class="MuiGrid-root css-1bb64vp-MuiGrid-root"
        imagesource="https://s3.ap-south-1.amazonaws.com/images-cms.uat.shopper-stop.in/Top_Brands_Mobile_84f472044f.png?updated_at=2022-10-03T14:23:49.866Z"
      >
        <div
          class="MuiBox-root css-1bpan0e"
        >
          <img
            class="css-7nqtc1"
            src="https://images-cms.uat.shopper-stop.in/Smashbox_bf109553e7.jpg"
          />
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-1l4w6pd"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-fwm6to-MuiButtonBase-root-MuiButton-root"
        imagesource="https://s3.ap-south-1.amazonaws.com/images-cms.uat.shopper-stop.in/Top_Brands_Mobile_84f472044f.png?updated_at=2022-10-03T14:23:49.866Z"
        tabindex="0"
        type="button"
      >
        EXPLORE BRANDS
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      
    </div>
  </div>
</div>
`;
